libdnf/repo/solvable/RelationalDependency.cpp

```diff
@@ -0,0 +1,104 @@
+#include <cassert>
+#include <solv/pool.h>
+
+#include "RelationalDependency.hpp"
+
+int RelationalDependency::determineComparisonType(const std::string &type)
+{
+    if (type == "<=") {
+        return REL_LT | REL_EQ;
+    } else if (type == ">=") {
+        return REL_GT | REL_EQ;
+    } else if (type == "==") {
+        return REL_EQ;
+    } else if (type == ">") {
+        return REL_GT;
+    } else if (type == "<") {
+        return REL_LT;
+    }
+
+    throw "Unknown comparison: " + type;
+}
+
+RelationalDependency::RelationalDependency() : pool(nullptr), id(-1)
+{}
+
+RelationalDependency::RelationalDependency(const RelationalDependency &src)
+        : pool(src.pool), id(src.id)
+{}
+
+RelationalDependency::RelationalDependency(Pool *pool, const std::string &name, const std::string &evr,
```

Comment:
this won't work for rich deps

Reply:
Well, for now you can always do:
```
auto richDep = RelationalDependency(pool, pool_parserpmrichdep(pool, richDependency));
```

Reply:
yes, but the whole point of libdnf is to NOT expose libsolv

Reply:
I made changes so it also detects rich deps without exposing libsolv